test/Interop/Cxx/function/Inputs/default-arguments.h

```diff
@@ -0,0 +1,55 @@
+inline void unused(int x = 0) {}
+
+inline bool isZero(int value = 0) { return value == 0; }
+inline bool isNonZero(int value = 0) { return value != 0; }
+
+inline bool isNil(int *ptr = nullptr) { return ptr == nullptr; }
+inline bool isStrNil(const char *ptr = "") { return ptr == nullptr; }
+
+inline int sum(int a, int b = 1) { return a + b; }
+inline int subtract(int a = 123, int b = 1) { return a - b; }
+
+static int counter = 0;
+inline bool isZeroCounter(int value = counter++) { return value == 0; }
+
+struct ArgTy {
+  int value;
+
+  static ArgTy createNonZero() { return {1}; }
+};
+inline bool isArgZero(ArgTy a = {0}) { return a.value == 0; }
+inline bool isArgNonZero(ArgTy a = ArgTy::createNonZero()) { return a.value != 0; }
+inline bool isArgZeroConstRef(const ArgTy &a = {0}) { return a.value == 0; }
+
+struct HasMethodWithDefaultArg {
+  bool isZero(int v = 0) const { return v == 0; }
+};
+
+struct DerivedFromHasMethodWithDefaultArg : public HasMethodWithDefaultArg {};
+
+struct HasStaticMethodWithDefaultArg {
+  static int value;
+  static bool isNonZero(int v = value);
+
+  static int counter;
+  static bool isNonZeroCounter(int v = counter++);
+
+private:
+  static int privateCounter;
+public:
+  static bool isNonZeroPrivateCounter(int v = privateCounter++);
+};
+
+// TODO: support default arguments of constructors
+// (https://github.com/apple/swift/issues/70124)
+struct HasCtorWithDefaultArg {
+  int value;
+
+  HasCtorWithDefaultArg(int a, int b = 456, int c = 123) : value(a + b + c) {}
+};
+
+inline int ambiguous(int a, int b = 1) { return a; }
+inline int ambiguous(int a) { return a; }
+
+inline int nonTrailing(int a, int b = 2) { return a + b; }
+inline int nonTrailing(int a = 1, int b);
```

Comment:
Can you add some additional tests for a class template, including a dependent value in a default parameter?

Reply:
Done